src/webui/api/torrentscontroller.cpp

```diff
@@ -241,7 +241,7 @@ namespace
 void TorrentsController::infoAction()
 {
     const QString filter {params()["filter"]};
-    const QString category {params()["category"]};
+    const QString category {(params().contains("category") && params()["category"].isNull()) ? QLatin1String("") : params()["category"].trimmed()};
```

Comment:
1. This line is too long, try applying a soft limit at 80~100 chars.
2. Why do you add `trimmed()`? I can use a white space as category name in GUI currently.

Reply:
To address point 2: everywhere else in this file that a `category` variable is initialized from `params()["category"]`, the `trimmed()` method is used. So what is the correct way/intended behavior? Shouldn't we remove `trimmed()` everywhere else as well?

Reply:
I fixed the issue "upstream" in the request parser as @glassez suggested, but I kept the `trimmed()` for now, waiting for your opinion on that.

Reply:
IMO, most uses of "trimmed()" when processing incoming data are incorrect.

Reply:
>So what is the correct way/intended behavior? 

IMO we should allow give as much freedom as possible when accepting user inputs, i.e. I think white space as category name should be supported (unless there are technical reasons not to).

Reply:
@Chocobo1 @glassez So should I remove the other uses of `trimmed()` as well, from the other similar methods? Tough I must say I question the usefulness of allowing category names to have whitespace at the start and end. A user might accidentally enter a whitespace character at the start or end when defining a category name and then when they try to match to that category name, they will be surprised it doesn't work. Of course "they should have been more careful", but is this really the UX that we want?

Reply:
>Of course "they should have been more careful"

:+1:
>but is this really the UX that we want?

There are two cases:
1. We don't limit usage of leading/trailing whitespaces and this may cause inconvenience to "careless" users.
2. We forbid usage of leading/trailing whitespaces and this may cause inconvenience to users that want more freedom.

Since the result is about the same, I prefer that possible problems are caused by inaction, but not by action. So I choose the first option.

Reply:
Ok, I reverted the change and also removed every `trimmed()` usage for categories and tags.